### PR TITLE
New Adapter: yahoossp (Rebrand yssp to yahoossp keeping the yssp alias)

### PR DESCRIPTION
**Pull request by oath-jac**
We were told by legal that we'll go further with `yahoossp` name. The verizonmedia adapter will be removed at later stage.

### CODE REVIEW

**Review by VeronikaSolovei9 — DISMISSED** [review]
LGTM

File: config/config.go [hunk 1]
```diff
@@ -867,6 +867,7 @@ func SetupViper(v *viper.Viper, filename string) {
 	v.SetDefault("adapters.viewdeos.endpoint", "http://ghb.sync.viewdeos.com/pbs/ortb")
 	v.SetDefault("adapters.visx.endpoint", "https://t.visx.net/s2s_bid?wrapperType=s2s_prebid_standard:0.1.0")
 	v.SetDefault("adapters.vrtcal.endpoint", "http://rtb.vrtcal.com/bidder_prebid.vap?ssp=1804")
+	v.SetDefault("adapters.yahoossp.disabled", true)
```
Comment (guscarreon) [inline config/config.go:870]: Now that we have this entry, Do we still want to keep line 875?
```
875      v.SetDefault("adapters.yssp.disabled", true)
```

**Review by guscarreon — COMMENTED** [review]
Looks good but I posted questions in regards of `YSSP` and `yssp` entries that I could still find inside `openrtb_ext/bidders.go`, `exchange/adapter_builders.go` and `config/config.go` but I'm not sure whether or not you want to keep them there purposefully or not. Please let me know.

Also, do we still need files `openrtb_ext/imp_yssp.go`, `static/bidder-params/yssp.json` and `static/bidder-info/yssp.yaml`? We seem to no longer be using the `ExtImpYSSP` struct inside `adapters/yahoossp/yahoossp.go` and `ExtImpYahooSSP` seems to be a identical to `ExtImpYSSP` except for the name.
```
1 package openrtb_ext
2
3 // ExtImpYSSP defines the contract for bidrequest.imp[i].ext.yssp
4 type ExtImpYSSP struct {
5     Dcn string `json:"dcn"`
6     Pos string `json:"pos"`
7 }
openrtb_ext/imp_yssp.go
```
`static/bidder-info/yssp.yaml` and `static/bidder-info/yahoossp.yaml` seem to be identical files too.

File: openrtb_ext/bidders.go [hunk 2]
```diff
@@ -328,6 +329,7 @@ func CoreBidderNames() []BidderName {
 		BidderViewdeos,
 		BidderVisx,
 		BidderVrtcal,
+		BidderYahooSSP,
```
[the remark this answers is not in the document]
Comment (guscarreon) [reply openrtb_ext/bidders.go:332]: Please disregard my comment above.

File: openrtb_ext/bidders.go [hunk 3]
```diff
@@ -195,6 +195,7 @@ const (
 	BidderVisx              BidderName = "visx"
 	BidderViewdeos          BidderName = "viewdeos"
 	BidderVrtcal            BidderName = "vrtcal"
+	BidderYahooSSP          BidderName = "yahoossp"
```
[the remark this answers is not in the document]
Comment (guscarreon) [reply openrtb_ext/bidders.go:198]: Please disregard my comment above.

File: config/config.go (hunk 1 above)
Comment (guscarreon) [reply config/config.go:870]: Please disregard my comment above.

File: exchange/adapter_builders.go [hunk 4]
```diff
@@ -252,15 +252,16 @@ func newAdapterBuilders() map[openrtb_ext.BidderName]adapters.Builder {
...
 		openrtb_ext.BidderViewdeos:          adtelligent.Builder,
 		openrtb_ext.BidderVisx:              visx.Builder,
 		openrtb_ext.BidderVrtcal:            vrtcal.Builder,
+		openrtb_ext.BidderYahooSSP:          yahoossp.Builder,
 		openrtb_ext.BidderYeahmobi:          yeahmobi.Builder,
 		openrtb_ext.BidderYieldlab:          yieldlab.Builder,
 		openrtb_ext.BidderYieldmo:           yieldmo.Builder,
 		openrtb_ext.BidderYieldone:          yieldone.Builder,
-		openrtb_ext.BidderYSSP:              yssp.Builder,
+		openrtb_ext.BidderYSSP:              yahoossp.Builder,
```
[the remark this answers is not in the document]
Comment (guscarreon) [reply exchange/adapter_builders.go:264]: Please disregard my comment above.

**Comment by SyntaxNode** [comment]
Looks good to me. Please remove the `ExtImpYSSP` as the intent is for the ExtImpXXX structs to correlate to an adapter implementation, and now you have - rightfully so - `ExtImpYahooSSP`.

**Comment by SyntaxNode** [comment]
> In the java server we just declare the alias in the server config aliases property.

The alias functionality in PBS-Go is still in its infancy. We'll continue to make improvements in the near future. Don't expect to see the exact same approach from PBS-Java, we have other ideas in mind. The end goal is similar though in that no code is required for aliases.

**Comment by SyntaxNode** [comment]
Please also remove the `/params/race/banner.json` file. That race detection pattern has been removed and replaced with enhanced memory sharing checks in the json test framework.

**Review by guscarreon — APPROVED** [review]
LGTM

**Comment by SyntaxNode** [comment]
@oath-jac This is looking good. Please open a docs PR for the new yahoossp bidder, just as you did for yssp previously. We will not merge this PR until the docs PR is opened.

**Review by VeronikaSolovei9 — APPROVED** [review]
LGTM

**Comment by SyntaxNode** [comment]
Docs PR: https://github.com/prebid/prebid.github.io/pull/3283

**Pull request merged**